org.eclipse.winery.tools.deployablecomponents/src/main/java/org/eclipse/winery/tools/deployablecomponents/crawler/GithubCrawler.java

```diff
@@ -0,0 +1,232 @@
+            if (authenticated) {
+                connection.setRequestProperty("Authorization", "Basic " + authEncoded);
+            }
+
+            if (connection.getResponseCode() != HttpURLConnection.HTTP_OK) {
+                // 403 Forbidden might be rate limit
+                if (connection.getResponseCode() == HttpURLConnection.HTTP_FORBIDDEN) {
+                    String limitRemaining = connection.getHeaderField("X-RateLimit-Remaining");
+                    if (limitRemaining != null && Integer.parseInt(limitRemaining) == 0) {
+                        successful = false;
+                        if (isSearch) {
+                            rateLimitResetSearch = Instant.ofEpochSecond(Long.parseLong(connection.getHeaderField("X-RateLimit-Reset")));
+                            connection.disconnect();
+                            sleepSearchRateLimit();
+                        } else {
+                            rateLimitReset = Instant.ofEpochSecond(Long.parseLong(connection.getHeaderField("X-RateLimit-Reset")));
+                            connection.disconnect();
+                            sleepRateLimit();
+                        }
+                    } else {
+                        fails++;
+                        successful = false;
+                        System.err.println("An error occurred in a http get request: " + connection.getResponseCode() + " - " + connection.getResponseMessage() + ". Failed " + fails + ". time.");
+                        sleepOneMinute();
+                    }
+                } else {
+                    fails++;
+                    successful = false;
+                    System.err.println("An error occurred in a http get request: " + connection.getResponseCode() + " - " + connection.getResponseMessage() + ". Failed " + fails + ". time.");
+                    sleepOneMinute();
+                }
+            } else {
+                successful = true;
+            }
+        } while (!successful);
+
+        BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+
+        if (isInRateLimit && isSearch) {
+            rateLimitRemainingSearch = Integer.parseInt(connection.getHeaderField("X-RateLimit-Remaining"));
+            rateLimitResetSearch = Instant.ofEpochSecond(Long.parseLong(connection.getHeaderField("X-RateLimit-Reset")));
+            System.out.println("Search rate limit: " + rateLimitRemainingSearch);
+        } else if (isInRateLimit) {
+            rateLimitRemaining = Integer.parseInt(connection.getHeaderField("X-RateLimit-Remaining"));
+            rateLimitReset = Instant.ofEpochSecond(Long.parseLong(connection.getHeaderField("X-RateLimit-Reset")));
+            System.out.println("Rate limit: " + rateLimitRemaining);
+        }
+        String line;
+        while ((line = reader.readLine()) != null) {
+            responseBuilder.append(line).append("\n");
+        }
+        reader.close();
```

Comment:
If there is an error in between, the stream will never be closed. Hence, please use the `try (Reader reader = new Reader(...)) { ... }` syntax to ensure that the stream is always closed.